app/exawind/exawind.cpp

```diff
@@ -77,30 +77,56 @@ int main(int argc, char** argv)
                       << std::endl;
     }
 
-    MPI_Comm amr_comm =
-        exawind::create_subcomm(MPI_COMM_WORLD, num_awind_ranks, 0);
-    MPI_Comm nalu_comm = exawind::create_subcomm(
-        MPI_COMM_WORLD, num_nwind_ranks, psize - num_nwind_ranks);
-
     const YAML::Node doc(YAML::LoadFile(inpfile));
     const YAML::Node node = doc["exawind"];
     const std::string amr_inp = node["amr_wind_inp"].as<std::string>();
-    const std::string nalu_inp = node["nalu_wind_inp"].as<std::string>();
     std::filesystem::path fpath_amr_inp(amr_inp);
     const std::string amr_log =
         fpath_amr_inp.replace_extension(".log").string();
     std::ofstream out(amr_log);
 
+    const auto nalu_inps = node["nalu_wind_inp"].as<std::vector<std::string>>();
+    const int num_nwsolvers = nalu_inps.size();
+    if (num_nwind_ranks < num_nwsolvers) {
+        throw std::runtime_error(
+            "Number of Nalu-Wind ranks is less than the number of Nalu-Wind "
+            "instances. Please have at least one rank per instance.");
+    }
+    const int ranks_per_nw_solver = num_nwind_ranks / num_nwsolvers;
```

Comment:
@marchdf Do I understand this correctly, in that the user will provide total number of MPI ranks to run to nalu-wind on, and not ranks per nalu-wind instance ?

Reply:
Yes that's correct. Total number of ranks for NW solvers will be user specified. Those are equally divided up among the NW solvers.

Reply:
Rest of the PR looks good too!

Reply:
looks good! This approach should work for the upcoming milestone. In a future version we might need different resources for a blade vs a nacelle

Reply:
agreed. We will cross that bridge when we get there. Maybe it's smart and you specify a target number of cells per node and it allocates ranks to each solver according to how many cells the mesh has.

Reply:
Could we post the new format for running multiple nalu wind instances somewhere ? Maybe in the description of the PR ?

Reply:
Great point @ashesh2512 . This is the only thing that changes:
```
  nalu_wind_inp:
    - sphere-nwind-0.yaml
    - sphere-nwind-1.yaml
```
So it's a yaml list now. I also updated the reg test of this software to show this.